src/test/file.rs

```diff
@@ -158,3 +960,664 @@ pub fn load(filename: &str) -> Result<test::File, Box<dyn Error + Send + Sync>>
         }
     }
 }
+
+pub fn generate_number<T>(spec: &Specification<T>, max_attemps: u16) -> Option<T>
+where
+    T: num::Num
+        + rand::distributions::uniform::SampleUniform
+        + std::cmp::PartialOrd
+        + std::default::Default
+        + Clone
+        + PartialEq
+        + Display
+        + PartialOrd
+        + fmt::Debug,
+    Specification<T>: Checker<Item = T>,
+{
+    if spec.value.is_some() {
+        return spec.value.clone();
+    }
+
+    let mut rng = rand::thread_rng();
+    (0..max_attemps)
+        .map(|_| {
+            return match spec.one_of.as_ref() {
+                Some(vals) => vals
+                    .get(rng.gen_range(0..vals.len()))
+                    .unwrap_or(&T::default())
+                    .clone(),
+                None => rng.gen_range(
+                    spec.min.clone().unwrap_or_default()..spec.max.clone().unwrap_or_default(),
+                ),
+            };
+        })
+        .filter(|v| {
+            spec.check(v, &|_e, _a| "".to_string())
+                .into_iter()
+                .collect::<Validated<Vec<()>, String>>()
+                .is_good()
+        })
+        .nth(0)
+}
+
+pub fn generate_float(spec: &FloatSpecification, max_attemps: u16) -> Option<f64> {
+    if spec.value.is_some() {
+        return spec.value.clone();
+    }
+
+    let mut rng = rand::thread_rng();
+    (0..max_attemps)
+        .map(|_| {
+            return match spec.one_of.as_ref() {
+                Some(vals) => vals
+                    .get(rng.gen_range(0..vals.len()))
+                    .unwrap_or(&f64::default())
+                    .clone(),
+                None => rng.gen_range(
+                    spec.min.clone().unwrap_or_default()..spec.max.clone().unwrap_or_default(),
+                ),
+            };
+        })
+        .filter(|v| {
+            spec.check(v, &|_e, _a| "".to_string())
+                .into_iter()
+                .collect::<Validated<Vec<()>, String>>()
+                .is_good()
+        })
+        .nth(0)
+}
+
+pub fn generate_string(spec: &Specification<String>, max_attemps: u16) -> Option<String> {
+    const CHARSET: &[u8] = b"ABCDEFGHIJKLMNOPQRSTUVWXYZ\
+                            abcdefghijklmnopqrstuvwxyz\
+                            "; // 0123456789)(*&^%$#@!~";
+
+    if spec.value.is_some() {
+        return spec.value.clone();
+    }
+
+    let mut rng = rand::thread_rng();
+    let string_length: usize = rng.gen_range(1..50);
+
+    for _ in 0..max_attemps {
```

Comment:
seems to be a copy/paste error. They all say "max_attemps" instead of "max_attempts"

Reply:
Thanks